pyanaconda/ui/gui/spokes/software.py

```diff
@@ -560,24 +557,38 @@ def on_environment_activated(self, listbox, row):
         self.refreshAddons()
         self._addonListBox.show_all()
 
+    def on_checkbox_toggled(self, button, row):
+        # Select the addon. The button is already toggled.
+        self._select_addon_at_row(row, button.get_active())
+
     def on_addon_activated(self, listbox, row):
+        # Skip the separator.
         box = row.get_children()[0]
         if isinstance(box, Gtk.Separator):
             return
 
-        # GUI selections means that packages are no longer coming from kickstart
-        self._kickstarted = False
-
+        # Select the addon. The button is not toggled yet.
         button = box.get_children()[0]
-        addons = self._allAddons()
-        group = addons[row.get_index()]
+        self._select_addon_at_row(row, not button.get_active())
+
+    def _select_addon_at_row(self, row, is_selected):
+        # GUI selections means that packages are no longer coming from kickstart.
+        self._kickstarted = False
 
-        new_btn_val = not button.get_active()
+        # Activate the row.
+        with blockedHandler(row.get_parent(), self.on_addon_activated):
+            row.activate()
 
+        # Activate the button.
+        box = row.get_children()[0]
+        button = box.get_children()[0]
         with blockedHandler(button, self.on_checkbox_toggled):
-            button.set_active(new_btn_val)
+            button.set_active(is_selected)
 
-        self._mark_addon_selection(group, new_btn_val)
+        # Do the selection.
```

Comment:
I think comment "Mark the selection" is better here because this is really just marking what was selected between environments. The actual selection is done by `button.set_active(is_selected)`.